proxy/src/fully_qualified_authority.rs

```diff
@@ -5,9 +5,26 @@ use std::str::FromStr;
 
 use http::uri::Authority;
 
+/// A fully qualified authority according to Kubernetes service naming
+/// conventions.
 #[derive(Clone, Debug, Eq, Hash, PartialEq)]
 pub struct FullyQualifiedAuthority(Authority);
 
+/// A fully qualified authority, and whether or not the `Destination` service
+/// should be queried for that name.
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct NamedAddress {
+    pub name: FullyQualifiedAuthority,
+    pub use_destination_service: bool
+}
+
+impl NamedAddress {
+    #[inline]
+    pub fn without_trailing_dot(&self) -> &Authority {
+        self.name.without_trailing_dot()
```

Comment:
let's just drop this and have callers use `self.name.without_trailing_dot()` directly. In general we should avoid these forwarding methods unless they save a non-trivial effort on the caller side.